api/api.yaml

```diff
@@ -0,0 +1,619 @@
+    post:
+      tags:
+        - issuer
+        - frontend
+      operationId: IssuerPageNewCredentialFormPost
+      summary: Issues a new credentials and renders them to a page
+      requestBody:
+        required: true
+        content:
+          application/x-www-form-urlencoded:
+            schema:
+              $ref: '#/components/schemas/VCCreate'
+      responses:
+        '200':
+          description: Show the created credential on a rendered page
+          content:
+            text/html:
+              schema: 
+                type: object
+  /issuer/api/v1/creddetails/{id}:
+    get:
+      tags:
+        - issuer
+        - frontend
+      parameters:
+        - $ref: '#/components/parameters/Id'
+      operationId: IssuerPageCredentialDetails
+      summary: Show the credential details on a rendered page
+      responses:
+        '200':
+          description: Present the credential details in a rendered html page
+          content:
+            text/html:
+              schema: 
+                type: object
+  /issuer/api/v1/displayqrurl/{id}:
+    get:
+      tags:
+        - issuer
+        - frontend
+      parameters:
+        - $ref: '#/components/parameters/Id'
+      operationId: IssuerPageDisplayQRURL
+      summary: Show the credential encoded as a qr-code on a rendered html page 
+      responses:
+        '200':
+          description: Present the credential qr-code in a rendered html page
+          content:
+            text/html:
+              schema: 
+                type: object
+    
+  /issuer/api/v1/allcredentials:
+    get:
+      tags:
+        - issuer
+        - api
+      operationId: IssuerAPIAllCredentials
+      summary: Return the ids of all issued credentials
+      responses:
+        '200':
+          description: Returns the list of ids
+          content:
+            application/json:
+              schema:
+                type: array
+                items: 
+                  $ref: '#/components/schemas/CredentialId'
+  /issuer/api/v1/credential:
+    post:
+      tags:
+        - issuer
+        - api
+      operationId: CreateNewCredential
+      summary: Issues a new credentials and returns it
+      requestBody:
+        required: true
+        content:
+          application/x-www-form-urlencoded:
+            schema:
+              $ref: '#/components/schemas/VCCreate'
+      responses:
+        '200':
+          description: Return the created credential
+          content:
+            application/json:
+              schema: 
+                $ref: '#/components/schemas/VerifiableCredential'
+  /issuer/api/v1/credential/{id}:
+    get:
+      tags:
+        - issuer
+        - api
+      parameters:
+        - $ref: '#/components/parameters/Id'
+      operationId: IssuerAPICredential
+      summary: Return credential with the given ID
+      responses:
+        '200':
+          description: The credential
+          content:
+            text/plain:
```

Comment:
Why is the credential returned as ```text/plain```, when the actual content is a json object instead?

Reply:
This is coming from previous implementations of the credential repository, where there could be several types of credentials in serialized form in the DB, eg.:
- CBOR (COVID and others)
- VCs in JWT form
- VCs in JSON-LD

In our case we have (for the moment) only the VCs in JSON-LD, but the field in the table is a byte array. Unfortunately, I named the table field as "JSON", which is incorrect.
The idea is that the database stores the credentials in "raw" serialization format, and is the Wallet the one which encodes/decodes the credential from/to the raw format.
Also, the same string stored in the database is the one used for transmission between the participants.
Anyway, I will revisit this design and the mimetype.